frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseFastForwardParentToMatchBranchAction.java

```diff
@@ -35,6 +35,10 @@ public IEnhancedLambdaLogger log() {
   public void update(AnActionEvent anActionEvent) {
     super.update(anActionEvent);
 
+    if (!canBeUpdated(anActionEvent)) {
```

Comment:
Why don't we override the `update` method within `BaseProjectKeyAvailabilityAssuranceAction` and disable the action there, on a missing project? We could depend on inheritance instead of "IF-ology". 
Note that most of the actions already have the following check:
```
if (!presentation.isEnabledAndVisible()) {
  return;
}
```

Reply:
Because we must stop processing `update` method immediately not disable the action and processing rest of method anyway, so we can't do that in this way, or I missing something.
`presentation.isEnabledAndVisible` check isn't enough.

Reply:
Just thinking aloud... what if we made `update` final in this class and made it:
1. call `super.update` (just as now)
1. do `canBeUpdated` check (just as now)
1. can a template method to be filled up by subclasses (TBD)?

Reply:
This way there's no risk that someone forgets about calling `super.update` and/or `canBeUpdated` in their subclass `update` impl

Reply:
Great idea!